datamodel/src/main/java/com/hartwig/serve/datamodel/serialization/util/BackwardsCompatibilityUtil.java

```diff
@@ -0,0 +1,186 @@
+package com.hartwig.serve.datamodel.serialization.util;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.function.Function;
+
+import com.google.common.collect.Lists;
+import com.hartwig.serve.datamodel.ActionableEvent;
+import com.hartwig.serve.datamodel.Knowledgebase;
+import com.hartwig.serve.datamodel.KnownEvent;
+import com.hartwig.serve.datamodel.characteristic.ActionableCharacteristic;
+import com.hartwig.serve.datamodel.characteristic.ActionableCharacteristicComparator;
+import com.hartwig.serve.datamodel.characteristic.ImmutableActionableCharacteristic;
+import com.hartwig.serve.datamodel.fusion.ActionableFusion;
+import com.hartwig.serve.datamodel.fusion.ActionableFusionComparator;
+import com.hartwig.serve.datamodel.fusion.ImmutableActionableFusion;
+import com.hartwig.serve.datamodel.fusion.ImmutableKnownFusion;
+import com.hartwig.serve.datamodel.fusion.KnownFusion;
+import com.hartwig.serve.datamodel.fusion.KnownFusionComparator;
+import com.hartwig.serve.datamodel.gene.ActionableGene;
+import com.hartwig.serve.datamodel.gene.ActionableGeneComparator;
+import com.hartwig.serve.datamodel.gene.ImmutableActionableGene;
+import com.hartwig.serve.datamodel.gene.ImmutableKnownCopyNumber;
+import com.hartwig.serve.datamodel.gene.ImmutableKnownGene;
+import com.hartwig.serve.datamodel.gene.KnownCopyNumber;
+import com.hartwig.serve.datamodel.gene.KnownCopyNumberComparator;
+import com.hartwig.serve.datamodel.gene.KnownGene;
+import com.hartwig.serve.datamodel.gene.KnownGeneComparator;
+import com.hartwig.serve.datamodel.hotspot.ActionableHotspot;
+import com.hartwig.serve.datamodel.hotspot.ActionableHotspotComparator;
+import com.hartwig.serve.datamodel.hotspot.ImmutableActionableHotspot;
+import com.hartwig.serve.datamodel.hotspot.ImmutableKnownHotspot;
+import com.hartwig.serve.datamodel.hotspot.KnownHotspot;
+import com.hartwig.serve.datamodel.hotspot.KnownHotspotComparator;
+import com.hartwig.serve.datamodel.immuno.ActionableHLA;
+import com.hartwig.serve.datamodel.immuno.ActionableHLAComparator;
+import com.hartwig.serve.datamodel.immuno.ImmutableActionableHLA;
+import com.hartwig.serve.datamodel.range.ActionableRange;
+import com.hartwig.serve.datamodel.range.ActionableRangeComparator;
+import com.hartwig.serve.datamodel.range.ImmutableActionableRange;
+import com.hartwig.serve.datamodel.range.ImmutableKnownCodon;
+import com.hartwig.serve.datamodel.range.ImmutableKnownExon;
+import com.hartwig.serve.datamodel.range.KnownCodon;
+import com.hartwig.serve.datamodel.range.KnownCodonComparator;
+import com.hartwig.serve.datamodel.range.KnownExon;
+import com.hartwig.serve.datamodel.range.KnownExonComparator;
+
+import org.jetbrains.annotations.NotNull;
+
+public final class BackwardsCompatibilityUtil {
```

Comment:
You are absolutely right, and not your fault since I also didn't think of it. 

An advantage of my current solution would be that OncoAct can simply bump the datamodel version without making any code changes, but if there are allowed to do that, then they may as well make the code change itself. Have asked Lieke for input. 

Agree the only way for true backwards compatibility would be to write the data as CKB. We could duplicate the data completely or add a flag (OncoAct uses a SERVE database from their own process anyways, and could set the flag).

Reply:
The flag is a good idea. That's actually a pretty normal pattern (feature toggle) for this sort of thing, although I would have the default as CKB such that you "opt-in" to the new feature.

Reply:
After discussion with Lieke have decided to completely remove my attempt at creating backwards compatible. Its fine if this release brakes downstream components.